document S3 RBAC preference on v5.14+

donsizemore opened the pull request:
**What this PR does / why we need it**:

explicitly document v5.14+'s preference, to make it clear for administrators

**Which issue(s) this PR closes**:

- Closes #10707

**Special notes for your reviewer**:

none

**Suggestions on how to test this**:

(Jim) - I'd suggest no testing as this is just a doc update. I'd merge it now but I don't want to run anything during the break.

add/remove roles to/from EC2 instance in AWS when the Dataverse instance in question is configured to use an S3 bucket

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

no

**Is there a release notes update needed for this change?**:

no

**Additional documentation**:

Preview at https://dataverse-guide--11111.org.readthedocs.build/en/11111/installation/config.html#amazon-s3-storage-or-compatible

nightowlaz reviewed (commented):
Very clear, especially with the link to the AWS docs on instance profiles. Thanks for the tip on the local configs as well.

donsizemore commented:
@nightowlaz the link and most of the text was already there, just farther down and not as insistent. I moved it up top and changed "you may use RBAC" to "Dataverse WILL use RBAC, like it or not."

nightowlaz commented:
Yes, sorry, I saw that previous section in the diff code, but just emphasizing that it was good! =)

qqmyers commented on doc/sphinx-guides/source/installation/config.rst:
```diff
@@ -1093,6 +1093,8 @@ The Dataverse Software S3 driver supports multi-part upload for large files (ove
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+**Note:** As of version 5.14, if Dataverse is running in an EC2 instance it will prefer RBAC for S3, even if administrators configure Dataverse with programmatic access keys. This is preferential from a security perspective as there are no keys to rotate or have stolen. If you intend to assign a role to your EC2 instance, you will still need the ``~/.aws/config`` file to specify the region but yo ...
+
```
FWIW: From https://github.com/IQSS/dataverse/blob/637c21684d45d8d176cd13bec1481877311959df/src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java#L1203-L1210 I think you can still use a profile but RBAC wins over the default profile.

nightowlaz replied:
yea i believe that was our experience as we've been running with a profile for months, years, but SSM somehow added the instance profile with it's default role and that took priority

donsizemore replied:
@qqmyers an excellent catch, and thank you for pointing this out!

qqmyers commented on doc/sphinx-guides/source/installation/config.rst:
```diff
@@ -1093,6 +1093,8 @@ The Dataverse Software S3 driver supports multi-part upload for large files (ove
 First: Set Up Accounts and Access Credentials
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+**Note:** As of version 5.14, if Dataverse is running in an EC2 instance it will prefer RBAC for the S3 default profile, even if administrators configure Dataverse with programmatic access keys. This is preferential from a security perspective as there are no keys to rotate or have stolen. If you intend to assign a role to your EC2 instance, you will still need the ``~/.aws/config`` file to speci ...
```
Thanks for the update. Minor text edit suggestions below.

**Note:** As of version 5.14, if Dataverse is running in an EC2 instance it will prefer Role Based Access Control (RBAC) over the S3 default profile, even if administrators configure Dataverse with programmatic access keys. (Named profiles can still be used to override RBAC for specific stores.) RBAC is preferential from a security perspective as there are no keys to rotate or have stolen. If you intend to assign a role to your EC2 instance, you will still need the ``~/.aws/config`` file to specify the region but you need not generate credentials for the default profile. For more information please see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html

qqmyers commented on doc/sphinx-guides/source/installation/config.rst:
```diff
@@ -1093,6 +1093,8 @@ The Dataverse Software S3 driver supports multi-part upload for large files (ove
 First: Set Up Accounts and Access Credentials
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+**Note:** As of version 5.14, if Dataverse is running in an EC2 instance it will prefer RBAC for the S3 default profile, even if administrators configure Dataverse with programmatic access keys. Named profiles can still be used to override RBAC for specific datastores. This is preferential from a security perspective as there are no keys to rotate or have stolen. If you intend to assign a role to ...
```
Sorry - there was more changed than just the part in parens (which I was suggesting to keep). In line 1 I changed 'for the S3 default profile' to say 'over' instead of 'for', I spelled out RBAC the first time as Role Based Access Control, and in the third sentence I changed 'This' to 'RBAC' Since the sentence about named profiles made 'This' ambiguous.

qqmyers reviewed (approved):
Looks good.

ofahimIQSS commented:
merging PR - proof read and verified
